starlite/contrib/jwt/jwt_auth.py

```diff
@@ -0,0 +1,359 @@
+            response_status_code: An optional status code for the response. Defaults to '201 Created'.
+            token_expiration: An optional timedelta for the token expiration.
+            token_issuer: An optional value of the token 'iss' field.
+            token_audience: An optional value for the token 'aud' field.
+            token_unique_jwt_id: An optional value for the token 'jti' field.
+
+        Returns:
+            A [Response][starlite.response.Response] instance.
+        """
+        encoded_token = self.create_token(
+            identifier=identifier,
+            token_expiration=token_expiration,
+            token_issuer=token_issuer,
+            token_audience=token_audience,
+            token_unique_jwt_id=token_unique_jwt_id,
+        )
+        return Response(
+            content=response_body,
+            headers={self.auth_header: self.format_auth_header(encoded_token)},
+            media_type=response_media_type,
+            status_code=response_status_code,
+        )
+
+    def create_token(
+        self,
+        identifier: str,
+        token_expiration: Optional[timedelta] = None,
+        token_issuer: Optional[str] = None,
+        token_audience: Optional[str] = None,
+        token_unique_jwt_id: Optional[str] = None,
+    ) -> str:
+        """Create a Token instance from the passed in parameters, persists and returns it.
+
+        Args:
+            identifier: Unique identifier of the token subject. Usually this is a user ID or equivalent kind of value.
+            token_expiration: An optional timedelta for the token expiration.
+            token_issuer: An optional value of the token 'iss' field.
+            token_audience: An optional value for the token 'aud' field.
+            token_unique_jwt_id: An optional value for the token 'jti' field.
+
+        Returns:
+            The created token.
+        """
+        token = Token(
+            sub=identifier,
+            exp=(datetime.now(timezone.utc) + (token_expiration or self.default_token_expiration)),
+            iss=token_issuer,
+            aud=token_audience,
+            jti=token_unique_jwt_id,
+        )
+        return token.encode(secret=self.token_secret, algorithm=self.algorithm)
+
+    def format_auth_header(self, encoded_token: str) -> str:
+        """Format a token according to the specified OpenAPI scheme.
+
+        Args:
+            encoded_token: An encoded JWT token
+
+        Returns:
+            The encoded token formatted for the HTTP headers
+        """
+        if self.openapi_components.securitySchemes:
+            security = self.openapi_components.securitySchemes.get(self.openapi_security_scheme_name, None)
+            if isinstance(security, SecurityScheme):
+                return f"{security.scheme} {encoded_token}"
+        return encoded_token
+
+
+class JWTCookieAuth(Generic[UserType], JWTAuth[UserType]):
```

Comment:
This looks awfully similar to the session backend cookie config. Maybe we can/should abstract this?

Reply:
I was talking about the cookie fields. But I guess this level of abstraction might also be just fine.